soco/core.py

```diff
@@ -1490,6 +1494,70 @@ def get_item_album_art_uri(self, item):
         else:
             return None
 
+    @only_on_master
+    def set_sleep_timer(self, sleep_time_seconds):
+        """Sets the sleep timer.
+
+        Args:
+            sleep_time_seconds (`int/NoneType`): How long to wait before
+                turning off speaker in seconds, None to cancel a sleep timer.
+                Maximum value of 86399
+
+        Raises:
+            SoCoException: Upon errors interacting with Sonos controller
+            ValueError: Argument/Syntax errors
+
+        """
+        # Note: A value of None for sleep_time_seconds is valid, and needs to
+        # be preserved distinctly separate from 0. 0 means go to sleep now,
+        # which will immediately start the sound tappering, and could be a
+        # useful feature, while None means cancel the current timer
```

Comment:
Good with a comment to explain the difference between of and 0. I do however this we do the check just one bit better. In general (as you noticed) it can give some problems to check whether something is None, simply by checking if it evaluates to False, if there is another valid input that also evaluates to False (like 0). However, a slightly cleaner way to do is to simply check whether variable is None, that way the checks can be rewritten to:

```python
if sleep_time_seconds is None:
    sleep_time = ''
elif sleep_time_seconds != int(sleep_time_seconds):
    ....
else:
    ....
```